modules/kubernetes-addons/karpenter/variables.tf

```diff
@@ -0,0 +1,27 @@
+variable "helm_config" {
+  type        = any
+  description = "Helm provider config for the Karpenter"
+  default     = {}
+}
+
+variable "eks_cluster_id" {
+  type        = string
+  description = "EKS cluster Id"
+}
+
+variable "irsa_policies" {
+  type        = list(string)
+  description = "Additional IAM policies for a IAM role for service accounts"
+  default     = []
+}
+
+variable "manage_via_gitops" {
+  type        = bool
+  default     = false
+  description = "Determines if the add-on should be managed via GitOps."
+}
+
+variable "tags" {
+  type        = map(string)
+  description = "Common Tags for AWS resources"
+}
```

Comment:
Should we allow users to provide a provisioner yaml as module input? We can apply it for them and it can be managed in their GitOps repo if they want.

Reply:
I think this is up for further discussion. There are number of options that customer can use by creating multiple YAMLs for different provisioners. Let's take this outside this PR